core/src/main/java/nl/nn/adapterframework/jta/narayana/NarayanaConfigurationBean.java

```diff
@@ -67,7 +80,43 @@ private Properties readXmlFile(String propertyFileName) {
 	 * Populate all the jBossTS EnvironmentBeans.
 	 */
 	@Override
-	public void afterPropertiesSet() {
+	public void afterPropertiesSet() throws ObjectStoreException {
 		PropertiesFactory.setDelegatePropertiesFactory(new NarayanaPropertiesFactory());
+
+		// Set/Update the JdbcAccess value
+		final MetaObjectStoreEnvironmentBean jdbcStoreEnvironment = BeanPopulator.getDefaultInstance(MetaObjectStoreEnvironmentBean.class);
+		if(JDBCStore.class.getCanonicalName().equals(jdbcStoreEnvironment.getObjectStoreType())) {
+			jdbcStoreEnvironment.setJdbcAccess(getObjectStoreJndiName());
+		}
+	}
+
+	/**
+	 * Tests if we can find a (valid) DataSource. Takes ApplicationServer specific jndiPrefixes into account.
+	 * If the DataSource is valid it will be returned with the {@link PoolingDataSourceJDBCAccess} prefix.
+	 */
+	private String getObjectStoreJndiName() throws ObjectStoreException {
+		String objectStoreDatasource = AppConstants.getInstance().getProperty("transactionmanager.narayana.objectStoreDatasource");
+		if(StringUtils.isBlank(objectStoreDatasource)) {
+			throw new ObjectStoreException("no datasource name provided, please set property [transactionmanager.narayana.objectStoreDatasource]");
+		}
+
+		if(applicationContext == null) {
+			throw new ObjectStoreException("no ApplicationContext to retrieve DataSource from");
+		}
+
+		JndiContextPrefixFactory jndiContextFactory = applicationContext.getBean("jndiContextPrefixFactory", JndiContextPrefixFactory.class);
+		String jndiPrefix = jndiContextFactory.getContextPrefix();
+		String fullJndiName = (StringUtils.isNotEmpty(jndiPrefix)) ? jndiPrefix + objectStoreDatasource : objectStoreDatasource;
+
+		try {
+			DataSource dataSource = ObjectLocator.lookup(fullJndiName, null, DataSource.class);
+			log.info("found Narayana ObjectStoreDatasource [{}]", dataSource);
+
+			StringBuilder builder = new StringBuilder(PoolingDataSourceJDBCAccess.class.getCanonicalName());
```

Comment:
Is het wel nodig om hier een `StringBuilder` te gebruiken?
Ik zie nergens een loop of conditie, kan je het dat niet beter aan de compiler overlaten?